### PR TITLE
UI: Add a whitelist in i18next to fix the issue #675

### PR DESCRIPTION
**Pull request by ericsyh**
## Why for change
To fix https://github.com/pingcap-incubator/tidb-dashboard/issues/675.
The root cause of issue is we declare two languages as `zh-CN` and `en` in i18next while some situations browser will return `zh` (Under my case the browser is `Chrome 83` the OS is `Windows 10`) which is out of our language definition. 

## What did i do
I added a whitelist only for `zh-CN` and `en` in i18next to block other cases. With the whitelist i18next will fallback to `en` by default if browser return language out of our definition like `zh` or `en-US`.

## Test
Passed on my environment. Now i18next got `en` as `i18nextLng`

image

### CODE REVIEW

**Comment by ericsyh** [comment]
@baurine PTAL :)

**Comment by baurine** [comment]
So by this way, 'zh-HK' / 'zh-TW' / 'zh-SG' (https://www.andiamo.co.uk/resources/iso-language-codes/) will be fallback to 'en' as well, not sure whether i18next has a customized fallback strategy.

**Comment by ericsyh** [comment]
Seems no customized fallback provided in i18next but has [custom Detector](https://github.com/i18next/i18next-browser-languageDetector#adding-own-detection-functionality).  I tried it but do not figure it out  :(

**Comment by baurine** [comment]
According to the document: https://www.i18next.com/principles/fallback

1. Per default locals containing region or script will take a translation from the pure language file if not found. (https://www.i18next.com/principles/fallback#locals-resolving). It means 'zh-XX' will fallback to 'zh' automatically
1. We can define an object for `fallbackLng` (https://www.i18next.com/principles/fallback#fallback-language) like this:

    ```
    fallbackLng: { 
        'zh-HK': ['zh-CN', 'en'], 
        'zh-TW': ['zh-CN', 'en'],
        ...
    }
    ```

I prefer to use the way 1. We can change all places used by 'zh-CN' to 'zh'. (for example, zh-CN.yaml to zh.yaml), this can prevent 'zh' /  'zh-HK' / 'zh-TW' / 'zh-SG' / 'zh-XXX' except 'zh-CN' fallback to 'en' with no more configuration. (need test to confirm)

@breeswish , how do you think about it?

**Comment by baurine** [comment]
Or I can try the custom detector later.

**Comment by breeswish** [comment]
Yes, I also think solution 1 looks best 

https://github.com/pingcap-incubator/tidb-dashboard/pull/677#issuecomment-655906807

**Comment by baurine** [comment]
Thanks for your contribution! LGTM, I will have a test soon.

File: ui/lib/apps/DashboardSettings/translations/zh.yaml [hunk 1]
```diff
@@ -0,0 +1,5 @@
+dashboard_settings:
```
Comment (breeswish) [inline ui/lib/apps/DashboardSettings/translations/zh.yaml:1]: We no longer have this file, you can remove it

Comment (ericsyh) [reply ui/lib/apps/DashboardSettings/translations/zh.yaml:1]: I see, you removed dashboard setting to the user profile page in previous commit. I will remove it.

Comment (ericsyh) [reply ui/lib/apps/DashboardSettings/translations/zh.yaml]: Push a new commit https://github.com/pingcap-incubator/tidb-dashboard/pull/677/commits/436817df5977987d29bbfb89da4c03801a5db880 to remove the dashboard setting page.

**Comment by breeswish** [comment]
@all-contributors please add @ericsyh  for code

**Pull request merged**